diff --git a/dracut-functions b/dracut-functions
index d4caacf..399103d 100755
--- a/dracut-functions
+++ b/dracut-functions
@@ -83,7 +83,7 @@ inst_library() {
 # search in the usual places to find the binary.
 find_binary() {
     local binpath="/bin /sbin /usr/bin /usr/sbin" p
-    [[ -x $1 ]] && { echo $1; return 0; }
+    [[ ${1:0:1} == "/" && -x $1 ]] && { echo $1; return 0; }
     for p in $binpath; do
 	[[ -x $p/$1 ]] && { echo "$p/$1"; return 0; }
     done
