diff --git a/modules.d/99base/init b/modules.d/99base/init
index 36f1a07..6e47f9f 100755
--- a/modules.d/99base/init
+++ b/modules.d/99base/init
@@ -225,6 +225,10 @@ else
     cp /tmp/net.* /dev/.initramfs/ >/dev/null 2>&1
 fi
 
+# Avoid bug #512796
+ln -sf /proc/mounts /etc/mtab
+umount -t devpts -a
+
 exec switch_root "$NEWROOT" "$INIT" $initargs || {
     # davej doesn't like initrd bugs
     echo "Something went very badly wrong in the initrd.  Please "
diff --git a/modules.d/99base/install b/modules.d/99base/install
index bc2336a..1aa3bf4 100755
--- a/modules.d/99base/install
+++ b/modules.d/99base/install
@@ -1,6 +1,10 @@
 #!/bin/bash
 dracut_install mount mknod mkdir modprobe pidof sleep chroot \
     sed ls flock cp mv dmesg rm ln
+
+# Avoid bug #512796
+dracut_install umount
+
 if [ ! -e "${initdir}/bin/sh" ]; then
     dracut_install bash
     (ln -s bash "${initdir}/bin/sh" || :)
