diff --git a/dracut b/dracut
index f738a63..a177459 100755
--- a/dracut
+++ b/dracut
@@ -139,6 +139,9 @@ srcmods="/lib/modules/$kernel/"
 [ -n "$drivers_dir" ] && srcmods="$drivers_dir"
 export srcmods
 
+# On Gentoo /boot is not mounted automatically
+mount_boot $outfile
+
 if [[ -f $outfile && ! $force ]]; then
     echo "Will not override existing initramfs ($outfile) without --force"
     exit 1
diff --git a/dracut-functions b/dracut-functions
index d4caacf..162d816 100755
--- a/dracut-functions
+++ b/dracut-functions
@@ -342,4 +342,19 @@ instmods() {
     done
 } 
 
+mount_boot() {
+    bootdir="/boot"
+    if [[ "$1" =~ "$bootdir" ]] ; then
+	if ! egrep -q "[[:space:]]${bootdir}[[:space:]]" /proc/mounts; then
+	    if egrep -q "^[^#].+[[:space:]]${bootdir}[[:space:]]" /etc/fstab; then
+		if ! mount /boot; then
+		    dwarning "Failed to mount $bootdir"
+		else
+		    dinfo "$bootdir mounted successfully"
+		fi
+	    fi
+	fi
+    fi
+}
+
 # vim:ts=8:sw=4:sts=4:et
